File: Programmers/[92335] k진수에서 소수 개수 구하기/ONE/92335.java
class Solution {
    public int solution(int n, int k) {
        int answer = 0;

        String[] tokens = Integer.toString(n, k).replaceAll("0+", "0").split("0");

        for (String token : tokens)
            if (isPrime(Long.parseLong(token)))
                answer++;

        return answer;
    }

    private boolean isPrime(long num) {
        if (num == 1)
            return false;
        for (long i = 2; i < Math.sqrt(num); i++)
            if (num % i == 0)
                return false;
        return true;
    }
}

File: Programmers/[92335] k진수에서 소수 개수 구하기/KingDonggyu/92335.js
function solution(n, k) {
  const isPrimeNumber = (num) => {
    if (num === 1) {
      return false;
    }
    for (let i = 2; i * i <= num; i++) {
      if (num % i === 0) {
        return false;
      }
    }
    return true;
  };

  let answer = 0;

  n.toString(k).split('0').forEach((num) => {
    if (num) {
      isPrimeNumber(+num) && answer++;
    }
  });

  return answer;
}
